packages/components/src/Select.tsx

```diff
@@ -0,0 +1,263 @@
+ * @param [props.menuPlacement] - determines where the menu is placed
+ * @param [props.placeholder] - The placeholder value for the select
+ * @param [props.styles] - custom styling for the select
+ */
+export const Select = <
+  Value,
+  IsMulti extends boolean = false,
+  Option extends { value: Value; label?: string | undefined } = { value: Value; label: string | undefined }
+>(
+  props: SelectProps<Value, IsMulti, Option>
+) => {
+  const { options = [], value, ...otherProps } = props;
+
+  // Allows passing options as a list of values instead of options objects.
+  // For example:
+  // options: ['foo', 'bar']
+  // instead of:
+  // options: [{ value: 'foo' }, { value: 'bar' }]
+  //
+  // Cast as Options[] is because TS can't figure out how the `!_isObject(options[0])` condition affects the type of newOptions.
+  const newOptions = (
+    options && !_.isObject(options[0]) ? _.map((value) => ({ value }), options) : options
+  ) as Option[];
+
+  const findValue = (target: Value): Option | null => {
+    const foundOption = _.find({ value: target }, newOptions) as Option | undefined;
+    return foundOption || null;
+  };
+
+  const ParameterizedBaseSelect = BaseSelect as typeof BaseSelect<Value, Option, IsMulti, never>;
+  return <ParameterizedBaseSelect findValue={findValue} value={value} options={newOptions} {...otherProps} />;
+};
+
+export type GroupedSelectProps<
+  Value,
+  IsMulti extends boolean,
+  Option extends { value: Value; label?: string | undefined },
+  Group extends RSelectGroupBase<Option>
+> = Omit<BaseSelectProps<Value, Option, IsMulti, Group>, 'findValue' | 'options'> & {
+  options: Group[];
+};
+
+/**
+ * @param props - see {@link https://react-select.com/props#select-props}
+ * @param props.value - a member of an inner options object
+ * @param props.options - an object with toplevel pairs of label:options where label is a group label and options is an array of objects containing value:label pairs
+ * @param [props.id] - The HTML ID to give the form element
+ */
+export const GroupedSelect = <
+  Value,
+  IsMulti extends boolean = false,
+  Option extends { value: Value; label?: string | undefined } = { value: Value; label?: string | undefined },
+  Group extends RSelectGroupBase<Option> = RSelectGroupBase<Option>
+>({
+  value,
+  options = [],
+  ...props
+}: GroupedSelectProps<Value, IsMulti, Option, Group>) => {
+  // cast because types don't carry through Lodash
+  const flattenedOptions = _.flatMap('options', options) as Option[];
+  const findValue = (target: Value) => (_.find({ value: target }, flattenedOptions) || null) as Option | null;
+
+  const ParameterizedBaseSelect = BaseSelect as typeof BaseSelect<Value, Option, IsMulti, Group>;
+  return <ParameterizedBaseSelect findValue={findValue} value={value} options={options} {...props} />;
+};
+
+export const AsyncCreatableSelect = (props) => {
+  return <RAsyncCreatableSelect {..._.merge(getCommonSelectProps, props)} />;
```

Comment:
`getCommonSelectProps` is a function that returns the props, so its return value needs to be merged with the `props` passed in.

```suggestion
  const theme = useThemeFromContext();
  return <RAsyncCreatableSelect {..._.merge(getCommonSelectProps(theme), props)} />;
```